src/Symfony/Bundle/SecurityBundle/Security/FirewallMap.php

```diff
@@ -60,15 +58,22 @@ public function getFirewallConfig(Request $request)
         return $context->getConfig();
     }
 
+    /**
+     * @param Request $request
+     *
+     * @return \Symfony\Bundle\SecurityBundle\Security\FirewallContext
+     */
     private function getFirewallContext(Request $request)
     {
-        if ($this->contexts->contains($request)) {
-            return $this->contexts[$request];
+        if ($request->attributes->has('_firewall_context')) {
```

Comment:
you should validate that `$request->attributes->get('_firewall_context')` is a valid key of `$this->map`